src/logic.js

```diff
@@ -121,7 +124,7 @@ const checkTransactionJson = (data) => {
 
   /* Checking the keys in the payload */
   const numKeys = Object.keys(payload).length;
-  if (numKeys < 8) return false;
+  if (numKeys < expectedFields.length) return false;
```

Comment:
I believe `@zilliqa-js/utils` has a function for testing if the payload is correct.